src/TransferRestrictor.sol

```diff
@@ -31,8 +30,13 @@ contract TransferRestrictor is AccessControlDefaultAdminRules, ITransferRestrict
 
     /// ------------------ Initialization ------------------ ///
 
-    constructor(address owner) AccessControlDefaultAdminRules(0, owner) {
-        _grantRole(RESTRICTOR_ROLE, owner);
+    function initialize(address initialOwner) public initializer {
+        __AccessControlDefaultAdminRules_init_unchained(0, initialOwner);
+        _grantRole(RESTRICTOR_ROLE, initialOwner);
+    }
+
+    function reinitialize(address initialOwner, address upgrader, string memory newVersion) external reinitializer(2) {
```

Comment:
This should not need a reinitialize method since this wil be upgradeable for the first time.